accounts/abi/type.go

```diff
@@ -96,64 +96,70 @@ func NewType(t string) (typ Type, err error) {
 		t += "256"
 	}
 
-	if isslice {
-		typ.Kind = reflect.Slice
-		typ.Size = size
-		switch vtype {
-		case "int":
-			typ.Type = big_ts
-		case "uint":
-			typ.Type = ubig_ts
-		default:
-			return Type{}, fmt.Errorf("unsupported arg slice type: %s", t)
-		}
-	} else {
-		switch vtype {
-		case "int":
-			typ.Kind = reflect.Ptr
-			typ.Type = big_t
-			typ.Size = 256
-			typ.T = IntTy
-		case "uint":
-			typ.Kind = reflect.Ptr
-			typ.Type = ubig_t
-			typ.Size = 256
-			typ.T = UintTy
-		case "bool":
-			typ.Kind = reflect.Bool
-			typ.T = BoolTy
-		case "real": // TODO
-			typ.Kind = reflect.Invalid
-		case "address":
+	/*
+		if isslice {
 			typ.Kind = reflect.Slice
-			typ.Type = address_t
-			typ.Size = 20
-			typ.T = AddressTy
-		case "string":
-			typ.Kind = reflect.String
-			typ.Size = -1
-			typ.T = StringTy
-			if vsize > 0 {
-				typ.Size = 32
+			typ.Size = size
+			switch vtype {
+			case "int":
+				typ.Type = big_ts
+			case "uint":
+				typ.Type = ubig_ts
+			default:
+				return Type{}, fmt.Errorf("unsupported arg slice type: %s", t)
 			}
-		case "hash":
-			typ.Kind = reflect.Slice
+		} else {
+	*/
+	switch vtype {
+	case "int":
+		typ.Kind = reflect.Ptr
+		typ.Type = big_t
+		typ.Size = 256
+		typ.T = IntTy
+	case "uint":
+		typ.Kind = reflect.Ptr
+		typ.Type = ubig_t
+		typ.Size = 256
+		typ.T = UintTy
+	case "bool":
+		typ.Kind = reflect.Bool
+		typ.T = BoolTy
+	case "real": // TODO
+		typ.Kind = reflect.Invalid
+	case "address":
+		typ.Type = address_t
+		typ.Size = 20
+		typ.T = AddressTy
+	case "string":
+		typ.Kind = reflect.String
+		typ.Size = -1
+		typ.T = StringTy
+		if vsize > 0 {
 			typ.Size = 32
-			typ.Type = hash_t
-			typ.T = HashTy
-		case "bytes":
-			typ.Kind = reflect.Slice
-			typ.Type = byte_ts
-			typ.Size = vsize
-			if vsize == 0 {
-				typ.T = BytesTy
-			} else {
-				typ.T = FixedBytesTy
-			}
-		default:
-			return Type{}, fmt.Errorf("unsupported arg type: %s", t)
 		}
+	case "hash":
+		typ.Kind = reflect.Array
+		typ.Size = 32
+		typ.Type = hash_t
+		typ.T = HashTy
+	case "bytes":
+		typ.Kind = reflect.Array
+		typ.Type = byte_ts
+		typ.Size = vsize
+		if vsize == 0 {
+			typ.T = BytesTy
+		} else {
+			typ.T = FixedBytesTy
+		}
+	default:
+		return Type{}, fmt.Errorf("unsupported arg type: %s", t)
 	}
+
+	if isslice {
```

Comment:
Might be worth a comment that this just modifies the previously determined types. Took me a while to realize that this code refines the things already set above.